packages/base/Collapse/src/Collapse/Collapse.tsx

```diff
@@ -28,70 +40,96 @@ export const Collapse = forwardRef<HTMLDivElement, Props>(
       unmountOnExit,
       style,
       appear,
-      onExited: onExitedProp,
       'data-testid': dataTestId,
       ...rest
     },
     ref
   ) => {
-    const collapseRef = useRef<HTMLDivElement>(null)
-    const combinedRef = useMultipleForwardRefs([ref, collapseRef])
+    const wrapperRef = useRef<HTMLDivElement>(null)
+    const nodeRef = useRef<HTMLDivElement>(null)
+    const combinedRef = useMultipleForwardRefs<HTMLDivElement>([ref, nodeRef])
 
-    const [height, setHeight] = useState<number>()
+    // Memoized style object to avoid unnecessary recalculations
+    const transitionStyle = useMemo(
+      () => ({ transitionDuration: `${timeout}ms`, ...style }),
+      [timeout, style]
+    )
 
-    const setToCurrentHeight = () => {
-      if (collapseRef.current) {
-        setHeight(collapseRef.current.scrollHeight)
-      }
+    const getWrapperSize = () =>
+      wrapperRef.current ? wrapperRef.current.clientHeight : 0
```

Comment:
This doesn't need to be a function, just a variable should work
```suggestion
    const wrapperSize = wrapperRef.current?.clientHeight ?? 0
```